network-api/networkapi/templates/pages/buyersguide/category_page.html

```diff
@@ -24,14 +24,23 @@ <h2>({{ category.name }}) Primary related articles (First 3 of list of 6)</h2>
               {% include "fragments/buyersguide/related_reading.html" with articles=primary_related_articles index_page=editorial_content_index %}
             </div>
 
-            {% with cta=featured_cta %}
-              {% if cta %}
-                {% include "fragments/buyersguide/call_to_action_box.html" with icon=cta.sticker_image heading=cta.title body=cta.content link_text=cta.link_label link_href=cta.get_target_url %}
-              {% endif %}
-            {% endwith %}
-            
           </div>
         {% endif %}
+
+        <div
+        class="
+          {% if category != current_category %}
+            d-none
+          {% endif %}
+        "
+        >
+          {% with cta=featured_cta %}
+            {% if cta %}
+              {% include "fragments/buyersguide/call_to_action_box.html" with icon=cta.sticker_image heading=cta.title body=cta.content link_text=cta.link_label link_href=cta.get_target_url %}
+            {% endif %}
+          {% endwith %}
+        </div>
+
```

Comment:
Have you considered also moving it outside of the `{% with primary_related_articles=category.get_primary_related_articles  %}` block? Seems like this is no information we need.

Also, it seems to me the `{% if cta %}` should be `{% if category.show_cta %}`.

Reply:
Hi Tibor, I have moved it out of the with block, as well as updated it to use category.show_cta.

Originally, we were checking if this was set to true in the backend and updating the context, but I think your solution makes more sense to me!